cmd/common/registry.go

```diff
@@ -237,9 +239,23 @@ func buildCommandFromGadget(
 			// the flags
 			checkVerboseFlag()
 
-			if c, ok := gadgetDesc.(gadgets.GadgetDescCustomParser); ok {
+			err := runtime.Init(runtimeGlobalParams)
+			if err != nil {
+				return fmt.Errorf("initializing runtime: %w", err)
+			}
+			defer runtime.Close()
+
+			var gadgetInfo *runTypes.GadgetInfo
+			// Before running the gadget, we need to get the gadget info to create the
+			// parser based on it
+			if isRunGadget {
 				var err error
-				parser, err = c.CustomParser(gadgetParams, cmd.Flags().Args())
+				gadgetInfo, err = runtime.GetGadgetInfo(gadgetDesc, gadgetParams, args)
```

Comment:
> This should also receive the context from fe.GetContext(). In this way, when it is using the gRPC runtime, if it gets stuck trying to connect to the pod, it will be cancelled if users press Ctrl+C

I agree. 

> Being fuzzy, it should also receive the timeout, but we can leave it for later and maybe use a context.WithValue() to pass the context always together with the timeout.

This timeout is different, it's handled in https://github.com/inspektor-gadget/inspektor-gadget/blob/d6f5582e292bb66be81daea9ae396872f072c620/pkg/runtime/grpc/grpc-runtime.go#L203

Reply:
> This timeout is different
>
That's my point. Why is it different? If users run `ig run --timeout 10`, and `ig` gets stuck trying to connect to the pod, it will return a timeout error after `ConnectTimeout` sec instead of after `10` sec (as the user requested).

Reply:
`--timeout` for the gadget means "run the gadget for X seconds", the timeout on the connection is "try to connect for X seconds". Something like `ig run foo --timeout 300` doesn't mean the user wants to wait 300 seconds for getting an error on the connection.